File: examples/read-results.mjs
#!/usr/bin/env node
/** Read the committed flat benchmark summary with Node.js built-ins. */
import { readFileSync } from 'node:fs';

const csvPath = process.argv[2] ?? 'benchmark/final-results.csv';
const [headerLine, ...dataLines] = readFileSync(csvPath, 'utf8').trim().split('\n');
const headers = headerLine.split(',');
const rows = new Map(
  dataLines.map((line) => {
    const row = Object.fromEntries(headers.map((header, index) => [header, line.split(',')[index]]));
    return [`${row.section}:${row.metric}`, row];
  }),
);

const recall = rows.get('quality:shortEventRecall');
const latency = rows.get('efficiency:latencyMs');
if (!recall || !latency) throw new Error('Expected result rows are missing from the CSV.');

console.log(JSON.stringify({
  source: csvPath,
  shortEventRecall: {
    agentic: Number(recall.agentic),
    static: Number(recall.static),
    agenticMinusStatic: Number(recall.agentic_minus_static),
  },
  latency: {
    agenticMs: Number(latency.agentic),
    staticMs: Number(latency.static),
    agenticVsStaticPercent: Number(latency.agentic_vs_static_percent),
  },
}, null, 2));

File: examples/read-results.py
#!/usr/bin/env python3
"""Read the committed flat benchmark summary with Python's standard library."""

import csv
import json
import sys
from pathlib import Path


def load_summary(path: Path) -> dict[str, object]:
    with path.open(newline="", encoding="utf-8") as handle:
        rows = {(row["section"], row["metric"]): row for row in csv.DictReader(handle)}

    recall = rows[("quality", "shortEventRecall")]
    latency = rows[("efficiency", "latencyMs")]
    return {
        "source": path.as_posix(),
        "shortEventRecall": {
            "agentic": float(recall["agentic"]),
            "static": float(recall["static"]),
            "agenticMinusStatic": float(recall["agentic_minus_static"]),
        },
        "latency": {
            "agenticMs": int(latency["agentic"]),
            "staticMs": int(latency["static"]),
            "agenticVsStaticPercent": float(latency["agentic_vs_static_percent"]),
        },
    }


csv_path = Path(sys.argv[1] if len(sys.argv) > 1 else "benchmark/final-results.csv")
print(json.dumps(load_summary(csv_path), indent=2))
